### ISSUE TITLE
M1 preview: Docker always starting – but nothing happens

### ISSUE THREAD

**Issue by macwinnie** [docker/for-mac#5187]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 4CBD2D75-9966-45CB-847C-8CFEAD36E9E9/20201224211910

### Expected behavior

* Docker starting
* Docker being usable

### Actual behavior

* Docker in starting mode for days ...

### Information
  - macOS Version: BigSur
  - MacBook Pro 13" Apple Silicone
  - Docker Preview linked in [Docker blog](https://www.docker.com/blog/download-and-try-the-tech-preview-of-docker-desktop-for-m1/)

### Diagnostic logs

```
Diagnose succeeded
```

### Steps to reproduce the behavior

1. Download preview 
2. Drag&Drop App to Application folder
3. Double Click App

Factory Reset, Login, ... nothing changed behaviour :(

**Comment by georgemathieson** [comment]
I was facing a similar issue, turns out I had checked the box to set the firewall to block all incoming connections.

Unchecking this box under firewall options fixed it for me.

**Comment by hcancelik** [comment]
> I was facing a similar issue, turns out I had checked the box to set the firewall to block all incoming connections.
> 
> Unchecking this box under firewall options fixed it for me.

Firewall is not enabled in my setup.

**Comment by snayski** [comment]
I'm having the same issue on an M1 Mac Mini. Docker initially worked and I was able to build and run one of my applications.

Then I changed the memory resource in the preferences from 2Gb to 1Gb; now Docker is stuck in the 'Starting' phase.

I tried rebooting the device and also reinstalling the Docker Preview to no avail.

In the system console, I see:

```
dial unix docker.raw.sock: connect: connection refused
still waiting for docker to respond after 30.094323542s
```
along with a bunch of errors from Docker and its associated processes about failing to access files due to permissions issues.

**Comment by snayski** [comment]
Update: I was able to fix this by selecting the 'Troubleshoot' button from the Docker menu:
<img width="358" alt="Screen Shot 2020-12-25 at 4 22 23 PM" src="https://user-images.githubusercontent.com/11650049/103142205-62842700-46cd-11eb-9d1b-a34653fc86e5.png">

And then selecting `Clean / Purge Data` followed by `Reset to factory defaults`:
<img width="1252" alt="Screen Shot 2020-12-25 at 4 22 47 PM" src="https://user-images.githubusercontent.com/11650049/103142211-73cd3380-46cd-11eb-9e58-cf7608f11841.png">

Not sure if this will work for you @macwinnie but it's worth a shot. There's definitely a bug here but hopefully this is a workaround for now until it is fixed.

**Comment by snayski** [comment]
Another update: I tried changing the memory resource to values other than 1 Gb and Docker was able to restart without any issues. It seems to only fail to restart when the value is changed to 1 Gb (1.25 Gb even worked).

**Comment by macwinnie** [comment]
Hi everyone =)

All options from within the Troubleshoot-Screen I already tried – changing the memory resource is not possible since that screen is blurred for me ...

Now, I've also given it a try and added Docker to the firewall exceptions manually even though my "Block all incoming connections" checkbox was not checked. That also did not work for me :(

Thanks for your support!
Best,
Martin

**Comment by colmr02** [comment]
hi everyone 
is anyone else having an issue where docker is failing while connected to vpn 

thanks for your help

2020-12-26T18:25:07Z dockerd time="2020-12-26T18:25:07.196073794Z" level=info msg="NetworkDB stats docker-desktop(22fea8747122) - netID:larqieedxrbqzst10r6d64369 leaving:false netPeers:1 entries:2 Queue qLen:0 netMsg/s:0"

**Comment by macwinnie** [comment]
Something strange happened tonight ... the Mac was rebooted multiple times with the error persisting ... today booting the mac and Docker is running without problems ...
Don't know, if that's a persisting state now or it'll be failing again soon.

**Comment by stephen-turner** [comment]
I'm going to close this ticket as the issue seems to have gone away.

**Comment by odeckmyn** [comment]
@stephen-turner just to let you know it still doesn't work for me either ("docker is starting" forever)
I'm using https://desktop.docker.com/mac/m1preview/Docker-AppleSilicon-Preview7.dmg 

My 2cts

**Comment by danielbryantuk** [comment]
I'm in the same situation with my M1 MacAir... "docker is starting" forever

I ran several containers when I initially installed preview 7, but soon as I (foolishly) enabled Kubernetes, I've been having no luck. I've disabled K8s, and tried all of the obvious resets/uninstall/restart etc, and still can't get Docker started

**Comment by stephen-turner** [comment]
OK, I'll reopen it. But please include a Diagnostics ID or logfiles, otherwise there's nothing for us to look at.

**Comment by vermaat** [comment]
Same here. Weirdly enough I was able to use the Docker Preview successfully on Big Sur 11.0 and a week after the update. Suddenly today it didn't start anymore, and I'm not aware of any changes to my system.

Big Sur 11.1
Macbook M1 Pro 8gb
Docker Preview M1 0.0.0 (1)

Diagnostics ID:     D1BCE827-C49A-435F-B779-B7A7C22050FF/20210106054458 (uploaded)

**Comment by macwinnie** [comment]
I've found that starting Docker at that direct moment, the Desktop is available and responsible after reboot, the Docker App regularly starts up successful.
Starting it automatically fails with my M1.
And starting some time after the reboot also fails.

**Comment by jnordberg** [comment]
Had the same issue. Seems like it's related to the network configuration. After disabling my VPN and wipe + factory reset it started up.

**Comment by macwinnie** [comment]
Hi everybody,

I've found (one?) reason for the M1 Docker preview not running smooth with at least my own M1 MacBook: The Karabiner Elements, witch somehow mess up the whole system ... :(

https://github.com/pqrs-org/Karabiner-Elements/issues/2517#issuecomment-779701035

Uninstalling the tool, the Docker preview runs smooth again ...

Best,
Martin

**Comment by odeckmyn** [comment]
> I've found (one?) reason for the M1 Docker preview not running smooth with at least my own M1 MacBook: The Karabiner Elements, witch somehow mess up the whole system ... :(

I'm not using it, and still don't have Docker running on my M1 system

**Comment by hcancelik** [comment]
There is a new version of the preview. Have you guys tried that? Not having the always starting issue anymore.

**Comment by odeckmyn** [comment]
@hcancelik oh ! you made my day !!! new version starts on my machine. Issue closed for me.

**Comment by danielbryantuk** [comment]
The new version works great on my M1 Macbook Air. Many thanks to the Docker team!

Link to the blog post/download page: https://www.docker.com/blog/new-docker-desktop-preview-for-apple-m1-released/

**Comment by ChrisLane** [comment]
I also experienced this. Docker Desktop preview wouldn't get past "starting..." until I disabled the "Block all incoming connections" option in the system firewall.

I was using version 3.1.0

**Comment by macwinnie** [comment]
Hi @ChrisLane,

for me, the problem is solved with version 3.1.0 – that's why I'm closing this specific issue.

If the problem still persists with your mac, would you mind to open a new issue with Diagnostics ID for your specific problem?

Thanks and best,
Martin

**Comment by jnordberg** [comment]
For me the problem persists with 3.1.0, docker won't start unless I kill it, turn off my VPN and start again
